lib/client.c

```diff
@@ -475,15 +475,32 @@ _snowflake_check_connection_parameters(SF_CONNECT *sf) {
     }
 
     char* top_domain = strrchr(sf->host, '.');
+    char host_without_top_domain[1024];
     if (top_domain)
     {
         top_domain++;
+        size_t length = top_domain - sf->host;
+        sf_strncpy(host_without_top_domain, sizeof(host_without_top_domain), sf->host,length);
     }
     else
     {
         // It's basically impossible not finding top domain in host.
         // Log the entire host just in case.
         top_domain = sf->host;
+        host_without_top_domain[0] = '\0';
+    }
+
+    //check privatelink
+    if (end_with(host_without_top_domain, PRIVATELINK_HOSTNAME_SUFFIX))
+    {
+        char url_buf[4096];
+        sf_sprintf(url_buf, sizeof(url_buf), "http://ocsp.%s/%s",
+           sf->host, "ocsp_response_cache.json");
+        log_trace(
```

Comment:
We need warning log if "SF_OCSP_RESPONSE_CACHE_SERVER_URL" has already been set (possibly by the user) and we change it to a different value. We could override user setting and we should notify them if we do so.

Reply:
I added a log message.